### ISSUE TITLE
Large python integer can't be handled with hooks

### ISSUE THREAD

**Issue by pramukta** [msgpack/msgpack-python#133]
This makes it impossible to work around an issue such as #71.  Here's a test that replicates the problem, based on the [example in the docs](https://github.com/msgpack/msgpack-python/blob/master/README.rst#packingunpacking-of-custom-data-type):

```python
 def test_overriding_hooks():
     def default(obj):
         if isinstance(obj, long):
             return {"__type__": "long", "__data__": str(obj)}
         else:
             return obj

     obj = {"testval": long(1823746192837461928374619)}
     refobj = {"testval": default(obj["testval"])}
     refout = msgpack.packb(refobj)
     assert isinstance(refout, str)
     testout = msgpack.packb(obj, default=default)

     assert refout == testout
```

The issue looks to be that the logic to run the default handler is executed at the end of the while block, after it has already tried to handle a bunch of types.  If any of those types match, the provided handler is not called.  In this case, the built-in handler errors out due to the number being too large ([which is expected](https://github.com/msgpack/msgpack/blob/master/spec.md#types-limitation)) and there is no way (that I can see) to extend the serialization to handle the case for a particular application.

I'm still trying to figure out a workaround/patch myself, but if anyone else has thoughts, it would be great to hear.   Other than preparing the data so that it doesn't have long integers (since I'm not in control of that).  Thanks for looking.

**Comment by pramukta** [comment]
Updating the title of this issue to be consistent with the pull request that addresses it (#135) so that there is a greater chance that it gets assessed / integrated.

**Comment by pramukta** [comment]
This issue was addressed via merging #135
